content/BindingMethods/BindingMethods.q.md

```diff
@@ -0,0 +1,1601 @@
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true, init=[commands/BindingMethods/DeleteJohnSmithStudentData.sql], filename=[connection.php,student.php]
+
+```php
+// connection.php
+<?php
+    $host = 'localhost';
+    $db = 'LibraryDB';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected to the database. <br />";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    $sql = "DELETE FROM students WHERE first_name = ? AND last_name = ?";
+
+    try {
+        $pstmt = $conn->prepare($sql);
+        $pstmt->bindValue(1, 'John');
+        $pstmt->bindValue(2, 'Smith');
+        if (!$pstmt->execute()) {
+            throw new Exception("Unable to delete student data in the table.");
+        }
+        echo "Successfully deleted the student data in the table.";
+    } catch (Execption $e) {
+        echo $e->getMessage();
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `Unable to delete student data in the table."`.
+
+- It prints `Successfully connected to the database.` and `Successfully deleted the student data in the table."`.
+
+
+/// type=SS, answer=[1]
+
+Which variable holds the SQL statement in `student.php`?
+
+- `$sql`
+
+- `$conn`
+
+- `$pstmt`
+
+- `last_name`
+
+- `first_name`
+
+
+/// type=SS, answer=[1]
+
+In `student.php`, which of the following is a positional placeholder as a parameter on line 4?
+
+- `?`
+
+- `$sql`
+
+- `students`
+
+- `DELETE FROM`
+
+- `first_name = ?`
+
+
+/// type=SS, answer=[4]
+
+In the statement `$pstmt->bindValue(1, 'John');` on line 8 of `student.php`, what is `1`?
+
+- It represents the explicit data type for the parameter.
+
+- It represents the parameter name as the parameter identifier.
+
+- It represents the value to bind to the positional placeholder `?` in the SQL statement. 
+
+- It represents the 1-index position of the positional placeholder `?` in the SQL statement.
+
+- It represents the variable to bind to the positional placeholder `?` in the SQL statement. 
+
+
+/// type=SS, answer=[4]
+
+In the statement `$pstmt->bindValue(1, 'John');` on line 8 of `student.php`, what does `John` represent?
+
+- The parameter name.
+
+- The explicit data type.
+
+- The 1-index position of the parameter.
+
+- The value to bind to the positional placeholder `?`.
+
+- The variable to bind to the positional placeholder `?`.
+
+
+/// type=MS, answer=[3,4]
+
+Which statements correctly describe `(1, 'John')` on line 8 of `student.php`?
```

Comment:
same here... Can we remove the parentheses `()` from `(1, 'John')`? Please check. Thanks.